### DISCUSSION TITLE
Can we build studio3 project on Windows 10 platform?

### DISCUSSION THREAD

**Discussion by GarfieldCatMao** [pawelsalawa/sqlitestudio#4009] [Q&A]
Hi There,

Great to see the latest version, very impressed.
We start our development on Studio this year, and we have been blocked by a building error for about two weeks.
Really hope that we can have your help.

Operating system:  Windows 10  (Both Enterprise and Family have this building error)
QT creator:  5.15.2 MinGW 8.1.0 64-bit
SQLite studio 3.30 source codes

We download sqlite3.dll, sqlite3.c and sqlite3.h manually from https://www.sqlite.org/index.html, since dependency file link this website provides is ineffective (All additional dependencies can be downloaded by clicking here (32-bit) or here (64-bit)).

We put *.dll and *.c/h in lib and include directions which are at the same level with Studio3(we just follow the instructions)

Then we start building project with QT creator, we expect to see a success, however we are blocked by an error.

Just curious that if failure is caused by absence of those additional dependencies?

Thanks

**Comment by pawelsalawa** [comment]
Hi,

Have you changed the compilation directory like mentioned here: https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_dev_setup_under_Windows
![image](https://user-images.githubusercontent.com/481846/109488985-8d507a00-7a86-11eb-9873-693905221902.png)

I know it is unusual to have one output directory for debug/release output, but that's something I've set up long time ago and I didn't really test (so far) if other configuration works.

**Reply by pawelsalawa** [reply]
Btw. You don't need sqlite3.dll nor sqlite3.h anymore. Version 3.3.0 already has SQLite3 source code included.

**Reply by GarfieldCatMao** [reply]
Hi Pawel,

Thanks quite a lot! We'll try it again.

BR,
Garfield

**Reply by yfdyh000** [reply]
> Btw. You don't need sqlite3.dll nor sqlite3.h anymore. Version 3.3.0 already has SQLite3 source code included.

This doesn't seem to apply to the latest master.

----
Changed to
```
debug:   DESTDIR = $$PWD/../$$OUTPUT_DIR_NAME/debug
release: DESTDIR = $$PWD/../$$OUTPUT_DIR_NAME/release
```
```
win32: {
    INCLUDEPATH += $$PWD/../include
    LIBS += -L$$PWD/../lib
}
```
in my dirs.pri file for building.

**Reply by pawelsalawa** [reply]
Right. Version 3.4.0 brings back the external dependency to sqlite3.dll (and sqlite3.h), because there was a request to be capable of using custom/updated SQLite3 version, even if SQLiteStudio doesn't update it.
